boa/contracts/abi/abi_contract.py

```diff
@@ -300,15 +293,16 @@ def at(self, address: Address | str) -> ABIContract:
         return contract
 
 
-def _abi_from_json(abi: dict) -> list | str:
+def _abi_from_json(abi: dict) -> str:
     """
-    Parses an ABI type into a list of types.
+    Parses an ABI type into its schema string.
     :param abi: The ABI type to parse.
-    :return: A list of types or a single type.
+    :return: The schema string for the given abi type.
     """
     if "components" in abi:
-        assert abi["type"] == "tuple"  # sanity check
-        return [_abi_from_json(item) for item in abi["components"]]
+        assert abi["type"] in ("tuple", "tuple[]")  # sanity check
+        components = [_abi_from_json(item) for item in abi["components"]]
+        return abi["type"].replace("tuple", f"({','.join(components)})")
```

Comment:
what is this line for?

Reply:
```python
if abi["type"] == "tuple": return f"(components)"
if abi["type"] == "tuple[]": return f"(components)[]"
return abi["type"]
```

Reply:
```suggestion
        components_str = ",".join(components)
        if abi["type"] == "tuple:
            return f"({components})"
        if abi["type"] == "tuple[]":
            return f"({components})[]"
        raise ValueError("not a tuple")
```

Reply:
```python
        match abi["type"]:
            case "tuple": return f"({components})"
            case "tuple[]": return f"({components})[]"
            case _: raise ValueError(f"Unknown type {abi['type']}")
```

Reply:
my only issue with this is that after linting we end up with extra indentation, but it's not really a big enough deal to be a blocker